src/vanilla/GeneratorSettingsJs.cs

```diff
@@ -42,5 +51,186 @@ public class GeneratorSettingsJs : IGeneratorSettings
         /// The folder where the generated files will be output to.
         /// </summary>
         public string OutputFolder { get; set; }
+
+        /// <summary>
+        /// If the PackageVersion property is null or empty, then first try to update it from an
+        /// existing package.json file in the output directory. If no existing package.json file
+        /// exists or it doesn't contain a version property, then update it to the latest version
+        /// found on NPM.
+        /// </summary>
+        public void UpdatePackageVersion()
+        {
+            IHost host = Settings.Instance?.Host;
+            if (string.IsNullOrEmpty(PackageVersion) && host != null)
+            {
+                string[] outputFolderContents = host.ListInputs("./").Result;
+                if (outputFolderContents == null || !outputFolderContents.Contains("package.json"))
+                {
+                    Log(Category.Information, "The output folder doesn't contain a pre-existing package.json file to get the package version from.");
+                }
+                else
+                {
+                    Log(Category.Information, "Getting the package version from the pre-existing package.json file in the output folder.");
+                    string packageJsonFileContents = host.ReadFile("package.json").Result;
+                    if (!string.IsNullOrEmpty(packageJsonFileContents))
+                    {
+                        try
+                        {
+                            JObject packageJson = JObject.Parse(packageJsonFileContents);
+                            string packageJsonVersion = packageJson["version"]?.ToString();
+                            if (string.IsNullOrEmpty(packageJsonVersion))
+                            {
+                                Log(Category.Information, "Pre-existing package.json file in the output folder didn't contain a non-empty version property.");
+                            }
+                            else
+                            {
+                                PackageVersion = packageJsonVersion;
+                                Log(Category.Information, $"Got version \"{PackageVersion}\" for package \"{PackageName}\" from existing package.json file.");
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            Log(Category.Error, $"{e.GetType().Name} - {e.Message}");

```

Comment:
If we have an exception here, do we want to continue?

Reply:
Yes. Getting the package version automatically is a nice feature, but if it fails for some reason it shouldn't cause the entire generation to stop. Logging an error message and having the version property in the package.json file is enough.